### ISSUE TITLE
assert self.summarizer_thread is None

### ISSUE THREAD

**Issue by paul-gauthier** [paul-gauthier/aider#182]
Discord user `infernix` getting this from time to time when editing code outside aider:

```
Traceback (most recent call last):
  File "/home/infernix/.local/bin/aider", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/infernix/.local/lib/python3.11/site-packages/aider/main.py", line 523, in main
    coder.run()
  File "/home/infernix/.local/lib/python3.11/site-packages/aider/coders/base_coder.py", line 339, in run
    new_user_message = self.run_loop()
                       ^^^^^^^^^^^^^^^
  File "/home/infernix/.local/lib/python3.11/site-packages/aider/coders/base_coder.py", line 410, in run_loop
    if self.should_dirty_commit(inp) and self.dirty_commit():
                                         ^^^^^^^^^^^^^^^^^^^
  File "/home/infernix/.local/lib/python3.11/site-packages/aider/coders/base_coder.py", line 893, in dirty_commit
    self.move_back_cur_messages(self.gpt_prompts.files_content_local_edits)
  File "/home/infernix/.local/lib/python3.11/site-packages/aider/coders/base_coder.py", line 393, in move_back_cur_messages
    self.summarize_start()
  File "/home/infernix/.local/lib/python3.11/site-packages/aider/coders/base_coder.py", line 368, in summarize_start
    assert self.summarizer_thread is None
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
```

**Comment by paul-gauthier** [comment]
Fixed in https://github.com/paul-gauthier/aider/commit/03ac5202dd141b19cb08351603e638d30cdffffd

**Comment by joshuavial** [comment]
Thanks for the fix, I've been seeing this come up a bit as well

**Comment by transcendr** [comment]
@paul-gauthier  excuse my ignorance I'm not really a python guy so still learning how the package manager, etc works. Could you tell me how I can get this fix? I installed via pip install and my current version say 0.11.0.

**Comment by paul-gauthier** [comment]
This is available in the main branch. You can get it by installing the latest version from github:

```
python -m pip install git+https://github.com/paul-gauthier/aider.git
```

I'll be pushing a bugfix release to pypi soon, hopefully later today.

**Comment by paul-gauthier** [comment]
This fix just went out in v0.11.1. I'm going to close this issue for now, but feel free to re-open or file a new issue any time.

**Comment by transcendr** [comment]
@paul-gauthier  thanks so much I got the update! Btw, aider is awesome. It's the first of these kinds of tools that is actually useful in a professional capacity. Keep it up, cheers!
